components/gitpod-cli/cmd/tasks-stop.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	supervisor_helper "github.com/gitpod-io/gitpod/gitpod-cli/pkg/supervisor-helper"
+	"github.com/gitpod-io/gitpod/supervisor/api"
+	supervisor "github.com/gitpod-io/gitpod/supervisor/api"
+	"github.com/manifoldco/promptui"
+	"github.com/spf13/cobra"
+)
+
+var stopTaskCmdOpts struct {
+	All bool
+}
+
+func stopTask(ctx context.Context, terminalClient supervisor.TerminalServiceClient, terminalAlias string) {
+	terminal, err := terminalClient.Get(context.Background(), &supervisor.GetTerminalRequest{Alias: terminalAlias})
+	if err != nil {
+		panic(err)
+	}
+
+	stopCmd := fmt.Sprintf("%s kill %d%s", "\003\n", terminal.Pid, "\n")
```

Comment:
```suggestion
	stopCmd := fmt.Sprintf("%s kill %d%s", "\003\n", terminal.Pid, "\n")
```

What this does:

1. `CTRL + C ` to stop potential foreground processes (eg. long running processes, watch, compilers, servers, etc)
2. Kill the task process by it's ID

`\003` -> `CTRL + C `
`\n` -> `Enter / Return`

Reply:
I wonder why do we need to connect to it and cannot send a system signal instead using something like `syscall.Kill`?

Reply:
@akosyakov I have replaced the approach from:

```go
stopCmd := fmt.Sprintf("%s kill %d%s", "\003\n", terminal.Pid, "\n")
_, err = terminalClient.Write(ctx, &api.WriteTerminalRequest{Alias: terminal.Alias, Stdin: []byte(stopCmd)})}
```

to:

```go
syscall.Kill(int(terminal.Pid), syscall.SIGINT)
syscall.Kill(int(terminal.Pid), syscall.SIGTERM)
```

I think code-wise looks better and more stable and and works similarly 👍

Reply:
Changed to use the `Shutdown` API.